lib/vdsm/network/link/setup.py

```diff
@@ -20,38 +20,74 @@
 from __future__ import absolute_import
 from __future__ import division
 
-from vdsm.network.link.bridge import Bridge
+
+class NmstateBridgeOpts:
+    def __init__(self):
+        self._types_dict = {
+            'mac-ageing-time': int,
+            'group-forward-mask': int,
+            'hash-max': int,
+            'multicast-snooping': self._str_to_bool,
+            'multicast-router': int,
+            'multicast-last-member-count': int,
+            'multicast-last-member-interval': int,
+            'multicast-membership-interval': int,
+            'multicast-querier': self._str_to_bool,
+            'multicast-querier-interval': int,
+            'multicast-query-use-ifaddr': self._str_to_bool,
+            'multicast-query-interval': int,
+            'multicast-query-response-interval': int,
+            'multicast-startup-query-count': int,
+            'multicast-startup-query-interval': int,
+        }
+
+    @property
+    def types_dict(self):
+        return self._types_dict
+
+    def convert_value(self, key, value):
+        conversion_func = self._types_dict[key]
+        return conversion_func(value)
+
+    @staticmethod
+    def _str_to_bool(str_value):
+        str_value = str_value.lower()
+        false_values = ['false', '0']
+        return str_value not in false_values
+
+
+class NmstateBridgeOptionNotSupported(Exception):
+    pass
 
 
 def parse_bond_options(options):
     """
     Parse bonding options into a dictionary.
     """
-
-    def _string_to_dict(str, div, eq):
-        if options == '':
-            return {}
-        return dict(
-            option.split(eq, 1) for option in str.strip(div).split(div)
-        )
-
     if options:
         d_options = _string_to_dict(options, ' ', '=')
         return d_options
     else:
         return {}
 
 
-def setup_custom_bridge_opts(nets):
-    for name, opts in parse_nets_bridge_opts(nets):
-        Bridge(name, opts)
+def parse_nets_bridge_opts(opts):
+
+    if not opts:
+        return {}
+    opts_str = opts.replace('_', '-')
+    bridge_opts = _string_to_dict(opts_str, ' ', '=')
+    try:
+        for key, value in bridge_opts.items():
+            bridge_opts[key] = NmstateBridgeOpts().convert_value(key, value)
```

Comment:
Now we are calling init for every loop run, please create it before the for as it is constant that can be reused.